galaxy_ng/app/api/v1/viewsets.py

```diff
@@ -0,0 +1,357 @@
+            return LegacyNamespace.objects.filter(name=name).order_by(order_by)
+
+        if keywords:
+            return LegacyNamespace.objects.filter(
+                Q(name__contains=keywords)
+            ).order_by(order_by)
+
+        return LegacyNamespace.objects.all().order_by(order_by)
+
+
+class LegacyRolesSetPagination(PageNumberPagination):
+    page_size = 100
+    page_size_query_param = 'page_size'
+    max_page_size = 1000
+
+
+class LegacyRoleViewSet(viewsets.ModelViewSet):
+    queryset = LegacyRole.objects.all().order_by('full_metadata__created')
+    serializer = LegacyRoleSerializer
+    serializer_class = LegacyRoleSerializer
+    permission_classes = [AllowAny]
+    pagination_class = LegacyRolesSetPagination
+    authentication_classes = [BasicAuthentication, SessionAuthentication, TokenAuthentication]
+
+    def get_queryset(self):
+
+        logger.info(f'QUERY_PARAMS: {self.request.query_params}')
+
+        keywords = None
+        if self.request.query_params.get('keywords'):
+            keywords = self.request.query_params.get('keywords').rstrip('/')
+
+        order_by = 'full_metadata__created'
+        if self.request.query_params.get('order_by'):
+            order_by = self.request.query_params.get('order_by').rstrip('/')
+            order_by = f'full_metadata__{order_by}'
+
+        github_user = None
+        for keyword in ['owner__username', 'github_user', 'namespace']:
+            if self.request.query_params.get(keyword):
+                github_user = self.request.query_params[keyword]
+                if github_user is not None:
+                    github_user = github_user.rstrip('/')
+                break
+
+        namespace = None
+        if github_user:
+            if github_user.isdigit():
+                namespace = LegacyNamespace.objects.filter(pk=int(github_user)).first()
+            else:
+                namespace = LegacyNamespace.objects.filter(name=github_user).first()
+
+        name = self.request.query_params.get('name')
+        if name is not None:
+            name = name.rstrip('/')
+
+        if github_user and name:
+            logger.info('FILTER BY USER AND NAME')
+            return LegacyRole.objects.filter(namespace=namespace, name=name).order_by(order_by)
+
+        elif github_user:
+            logger.info(f'FILTER BY USER: {github_user}')
+            logger.info(f'FILTER BY NAMESPACE: {namespace}')
+            return LegacyRole.objects.filter(namespace=namespace).order_by(order_by)
+
+        if keywords:
+            return LegacyRole.objects.filter(
+                Q(namespace__name__contains=keywords)
+                | Q(name__contains=keywords)
+                | Q(full_metadata__description__contains=keywords)
+            ).order_by(order_by)
+
+        # galaxy cli uses autocomplete kwargs for searching ...
+        autocomplete_kwargs = {
+            'autocomplete': None,
+            'platforms_autocomplete': None,
```

Comment:
is `platforms_autocomplete` handled below?

Reply:
I think I need to add platforms to the serializer first.